### PR TITLE
Add docs for form-layout and form-components

### PR DESCRIPTION
**Pull request by jamesricky**

### CODE REVIEW

**Review by nsams — APPROVED** [review]
:heart:

**Comment by nsams** [comment]
interesting, form is now the first entry (starting with layout)

**Comment by jamesricky** [comment]
> interesting, form is now the first entry in the left menu (starting with layout)

I adjusted the order.

File: packages/admin-stories/src/docs/form/Layout.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+# Form Layout
+
+Different types of layouts can be achieved, by adding `variant="horizontal"`, `variant="vertical"` or `fullWidth` to your field's props.
+
+When using `variant="vertical"` (which is the default value) the label is shown above the field. This is usually the most compact layout type.  
+With `variant="horizontal"` the label is shown on the left side of the field, where all labels have a fixed width.  
+Adding the `fullWidth` prop will cause the field and it's input to take up all the width, given by the parent element.
+
+<br />
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/form/Layout.stories.mdx:13]: Do we need Breaks in Markdown 🤔

File: packages/admin-stories/src/docs/form/Layout.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+const flavourOptions = [
+    { value: "chocolate", label: "Chocolate" },
+    { value: "strawberry", label: "Strawberry" },
+    { value: "vanilla", label: "Vanilla" },
+];
+
+storiesOf("stories/form/Layout", module)
+    .add("Fields in sidebar", () => {
+        const Root = styled.div`
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/form/Layout.stories.tsx:46]: Creating styled components inside the render function of a story may cause some performance issues (basically, this is the same as a React function component). I'd move the styles outside of the story.

Comment (nsams) [reply packages/admin-stories/src/docs/form/Layout.stories.tsx:46]: but then you won't see it in the source when switching the story into source mode

Comment (johnnyomair) [reply packages/admin-stories/src/docs/form/Layout.stories.tsx:46]: Good point! Then we'll leave it like this. Premature optimization... 😉

File: packages/admin-stories/src/docs/form/Layout.stories.mdx (hunk 1 above)
Comment (jamesricky) [reply packages/admin-stories/src/docs/form/Layout.stories.mdx:13]: We don't **need** them, but they make it look nicer 😄

If we strictly want to follow the markdown rules, we could probably add some custom css somewhere to make headlines have more margin or something...

**Review by kaufmo — APPROVED** [review]

File: packages/admin-stories/src/docs/form/Layout.stories.mdx (hunk 1 above)
Comment (jamesricky) [reply packages/admin-stories/src/docs/form/Layout.stories.mdx]: @johnnyomair I removed all `<br />` in markdown and styled the headlines instead.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**